### ISSUE TITLE
Unable to extract subset font name using the newer versions of PyMuPDF : 1.24.6 and 1.24.7.  

### ISSUE THREAD

**Issue by SirishaGorasa** [pymupdf/PyMuPDF#3677]
### Description of the bug

I am trying to extract the subset font name from span.

Although I set subset font names to TRUE, I can see the output as below.

**Calibri
MinionPro**

My expected output is as follows:
**AGDFCT+Calibri
THUGIK+MinionPro**

### How to reproduce the bug

This is the code that can help reproduce the bug
```
import fitz
from fitz import TOOLS
TOOLS.set_subset_fontnames(True)
doc=fitz.open("input.pdf")
for page in doc:
     for block in page.get_text("dict")["blocks"]:
            if block['type'] == 0:
                if 'lines' in block.keys():
                    for line in block["lines"]:
                        for span in line["spans"]:
                            font_name=span["font"]
                            print(font_name)
                         
doc.close()

```

### PyMuPDF version

1.24.6

### Operating system

Windows

### Python version

3.8

**Comment by JorjMcKie** [comment]
You did not supply a reproducing file.
This has not changed in the versions you mention. If you look at `page.get_fonts()` you will probably also see no subset flags in the font names.

**Comment by SirishaGorasa** [comment]
I am attaching a different file which has the same issue. When I use page.get_fonts(), I got the base font name with subset characters, but when I have used span["font'], the base font name didn't have subset characters.
[Encoding.pdf](https://github.com/user-attachments/files/16189253/Encoding.pdf)
I am providing the output of page.get_fonts() for this PDF.

![image](https://github.com/user-attachments/assets/5391ffd8-342a-427e-86a6-63145258d302)

The output of span["font'] is as below,

![image](https://github.com/user-attachments/assets/d476c87e-ccb2-44ea-a85d-b69f89a3e846)

**Comment by SirishaGorasa** [comment]
This is my PyMuPDF version 
![image](https://github.com/user-attachments/assets/fda266e7-825e-4df7-9154-daa1947284d3)

**Comment by JorjMcKie** [comment]
Thanks for the file - We will look into this.

**Comment by SirishaGorasa** [comment]
Sure, please

**Comment by JorjMcKie** [comment]
Located the problem.
As a circumvention (performance degradation!) do `pymupdf.g_use_extra=False` (right after the import or set/unset around the respective code portions). This will activate the internal pure Python code which does this correctly.

**Comment by SirishaGorasa** [comment]
> Located the problem. As a circumvention (performance degradation!) do `pymupdf.g_use_extra=False` (right after the import or set/unset around the respective code portions). This will activate the internal pure Python code which does this correctly.

Can you give an example snippet?

I have tried it the below way but couldn't see font names with subset characters.
```
import fitz
from fitz import TOOLS
from fitz import g_use_extra
g_use_extra=False

TOOLS.set_subset_fontnames(True)
doc=fitz.open("Encoding.pdf")
for page in doc:
     for block in page.get_text("dict")["blocks"]:
            if block['type'] == 0:
                if 'lines' in block.keys():
                    for line in block["lines"]:
                        for span in line["spans"]:
                            text=span["text"]
                            font_name=span["font"]
                            print(font_name)
                         
doc.close()

```

**Comment by JorjMcKie** [comment]
I used this:
```python
import pymupdf
from pprint import pprint

pymupdf.g_use_extra = False
print(pymupdf.version)
pymupdf.TOOLS.set_subset_fontnames(True)
doc = pymupdf.open("Encoding.pdf")
page = doc[0]
pprint(page.get_fonts(True))
spans = [
    s for b in page.get_text("dict")["blocks"] for l in b["lines"] for s in l["spans"]
]
pprint(spans)
```
And got this:
```python
('1.24.7', '1.24.4', '20240626000001')
[(24, 'ttf', 'Type0', 'BCDEEE+Aptos', 'C2_0', 'Identity-H', 0),
 (30, 'ttf', 'Type0', 'BCDGEE+Calibri-Light', 'C2_1', 'Identity-H', 0),
 (33, 'ttf', 'TrueType', 'BCDFEE+Aptos', 'TT0', 'WinAnsiEncoding', 0),
 (36, 'ttf', 'TrueType', 'BCDHEE+Calibri-Light', 'TT1', 'WinAnsiEncoding', 0)]
[{'ascender': 0.93896484375,
  'bbox': (72.0, 72.012451171875, 242.9677276611328, 86.660888671875),
  'color': 0,
  'descender': -0.28173828125,
  'flags': 4,
  'font': 'BCDEEE+Aptos',
  'origin': (72.0, 83.280029296875),
  'size': 12.0,
  'text': 'The text is in Identity-H encoding.'},
 {'ascender': 0.93896484375,
  'bbox': (243.02000427246094,
           72.012451171875,
           245.45750427246094,
           86.660888671875),
  'color': 0,
  'descender': -0.28173828125,
  'flags': 4,
  'font': 'BCDFEE+Aptos',
  'origin': (243.02000427246094, 83.280029296875),
  'size': 12.0,
  'text': ' '},
 {'ascender': 0.75,
  'bbox': (72.0, 99.3599853515625, 250.27700805664062, 111.3599853515625),
  'color': 0,
  'descender': -0.25,
  'flags': 0,
  'font': 'BCDGEE+Calibri-Light',
  'origin': (72.0, 108.3599853515625),
  'size': 12.0,
  'text': 'The text is in Windows Ansi encoding'},
 {'ascender': 0.75,
  'bbox': (250.27999877929688,
           99.3599853515625,
           252.99288940429688,
           111.3599853515625),
  'color': 0,
  'descender': -0.25,
  'flags': 0,
  'font': 'BCDHEE+Calibri-Light',
  'origin': (250.27999877929688, 108.3599853515625),
  'size': 12.0,
  'text': ' '}]
```

**Comment by SirishaGorasa** [comment]
Thanks a lot. 🙂
It worked.

**Comment by JorjMcKie** [comment]
I am glad it works for you!
But let's keep the issue open until the fix is available on PyPI - this is the policy we usually apply.

**Comment by SirishaGorasa** [comment]
Agreeable 
Sure, won't close the issue from my side until the fix is developed.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.8.

**Comment by SirishaGorasa** [comment]
I am facing the problem of not getting subset characters in the font name of the span using PyMuPDF 1.24.11.
I have tried using the below two codes 
```
import pymupdf
pymupdf.g_use_extra = False

pymupdf.TOOLS.set_subset_fontnames(True)
doc=pymupdf.open("ForresterAI.


pdf")
for page in doc:
    for block in page.get_text("dict")["blocks"]:
        if block['type'] == 0:
            if 'lines' in block.keys():
                for line in block["lines"]:
                    for span in line["spans"]:
                        print(span["font"])
```

```
import fitz
from fitz import TOOLS
fitz.TOOLS.set_subset_fontnames(True)
doc=fitz.open("ForresterAI.pdf")
for page in doc:
    for block in page.get_text("dict")["blocks"]:
        if block['type'] == 0:
            if 'lines' in block.keys():
                for line in block["lines"]:
                    for span in line["spans"]:
                        print(span["font"])
```
Both had given 5 different fonts like below. 

1. ProximaNova
2. ProximaNova-Bold
3. Helvetica-Bold
4. ProximaNova-Italic
5. ProximaNova-SemiBold

When I had tried printing fonts from page using `page.get_fonts(True)`, ProximaNova-Italic had subset characters to it but in span, there aren't any subset characters to it.

Can you let me know why is this ambiguity happening? I am attaching the file too.

[ForresterAI.pdf](https://github.com/user-attachments/files/18685751/ForresterAI.pdf)
